tests/test_hooks_integration.py

```
@@ -80,3 +80,99 @@ async def pre_input(self, request, body):
                     parent.rmdir()
         except Exception:
             pass
+
+
+def test_pre_input_hook_nested_mutation_propagates(tmp_path):
```

Ruff (0.13.2)

Unused function argument: `tmp_path`

(ARG001)